src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -735,6 +741,556 @@ def set_up_scale_rule(self):
         return scale_def
 
 
+class ContainerAppUpdateDecorator(BaseContainerAppDecorator):
+    def __init__(
+        self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str
+    ):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.containerapp_def = {}
+        self.new_containerapp = {}
+
+    def validate_arguments(self):
+        validate_revision_suffix(self.get_argument_revision_suffix())
+        # Validate that max_replicas is set to 0-1000
+        if self.get_argument_max_replicas() is not None:
+            if self.get_argument_max_replicas() < 1 or self.get_argument_max_replicas() > 1000:
+                raise ArgumentUsageError('--max-replicas must be in the range [1,1000]')
+
+    def update(self):
+        try:
+            r = self.client.update(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name(), container_app_envelope=self.new_containerapp,
+                no_wait=self.get_argument_no_wait())
+            if not self.get_argument_no_wait() and "properties" in r and "provisioningState" in r["properties"] and r["properties"]["provisioningState"].lower() == "waiting":
+                logger.warning('Containerapp update in progress. Please monitor the update using `az containerapp show -n {} -g {}`'.format(self.get_argument_name(), self.get_argument_resource_group_name()))
+            return r
+        except Exception as e:
+            handle_raw_exception(e)
+
+    def post_process(self, r):
+        # Delete managed bindings
+        linker_client = None
+        if self.get_argument_unbind_service_bindings():
```

Comment:
one more question, do we want to support the feature in cli core?

Reply:
the `--bind` has no preview_flag, but it is argument for `az containerapp service`, I will move this part to previewDecorator.
`Space separated list of services(bindings) to be connected to this app. e.g. SVC_NAME1[:BIND_NAME1] SVC_NAME2[:BIND_NAME2]...`